src/main/haskell/kore/src/Kore/Builtin/Map.hs

```diff
@@ -71,89 +89,245 @@ sortDeclVerifiers = HashMap.fromList [ (sort, Builtin.verifySortDecl) ]
 symbolVerifiers :: Builtin.SymbolVerifiers
 symbolVerifiers =
     HashMap.fromList
-    [
-      ("MAP.bind"
-      , Builtin.verifySymbol trivialVerifier
-          [ trivialVerifier
-          , trivialVerifier
-          , trivialVerifier
-          ]
+    [ ( "MAP.concat"
+      , Builtin.verifySymbol assertSort [assertSort , assertSort]
       )
-    , ("MAP.emp"
-      , Builtin.verifySymbol trivialVerifier
-          [ trivialVerifier
-          , trivialVerifier
-          , trivialVerifier
-          ]
+    , ( "MAP.element"
+      , Builtin.verifySymbol assertSort [anySort, anySort]
       )
-    , ("MAP.merge"
-      , Builtin.verifySymbol trivialVerifier
-          [ trivialVerifier
-          , trivialVerifier
-          ]
+    , ( "MAP.lookup"
+      , Builtin.verifySymbol anySort [assertSort, anySort]
       )
-    , ("MAP.element"
-      , Builtin.verifySymbol trivialVerifier
-          [ trivialVerifier
-          , trivialVerifier
-          ]
+    , ( "MAP.unit"
+      , Builtin.verifySymbol assertSort []
       )
-    , ("MAP.lookup"
-      , Builtin.verifySymbol trivialVerifier
-          [ assertSort
-          , trivialVerifier
-          ]
+    , ( "MAP.update"
+      , Builtin.verifySymbol assertSort [assertSort, anySort, anySort]
+      )
+    , ( "MAP.in_keys"
+      , Builtin.verifySymbol Bool.assertSort [anySort, assertSort]
       )
     ]
   where
-    trivialVerifier :: Builtin.SortVerifier
-    trivialVerifier = const $ const $ Right ()
-
-isHook
-    :: MetadataTools Object StepperAttributes
-    -> SymbolOrAlias Object
-    -> String
-    -> Bool
-isHook tools sym hookName =
-    hook (symAttributes tools sym) == Hook (Just hookName)
-
-evalBind :: Builtin.Function
-evalBind =
-    ApplicationFunctionEvaluator evalBind0
-  where
-    evalBind0 _ _ _ = notApplicableFunctionEvaluator
+    anySort :: Builtin.SortVerifier
+    anySort = const $ const $ Right ()
+
+type Builtin = Map (CommonPurePattern Object) (CommonPurePattern Object)
+
+{- | Abort function evaluation if the argument is not a Map domain value.
+
+    If the operand pattern is not a domain value, the function is simply
+    'NotApplicatible'. If the operand is a domain value, but not represented
+    by a 'BuiltinDomainMap', it is a bug.
+
+ -}
+expectBuiltinDomainMap
+    :: Monad m
+    => String  -- ^ Context for error message
+    -> CommonPurePattern Object  -- ^ Operand pattern
+    -> ExceptT (AttemptedFunction Object Kore.Variable) m Builtin
+expectBuiltinDomainMap ctx =
+    \case
+        Kore.DV_ _ domain ->
+            case domain of
+                Kore.BuiltinDomainMap _map -> return _map
```

Comment:
Same question as above: Why not map_ or map' or something else? The same for all other occurrences of _map, _key, _value and maybe others.